tests/test__collection_api.py

```diff
@@ -3136,6 +3136,111 @@ def test__aggregate_project_if_null_expression(self):
+                            {'case': 'this is true', 'then': 'first case'},
+                            {'case': False, 'then': 'second case'},
+                        ],
+                        'default': 'did not match',
+                    }
+                },
+                'missing_field': {
+                    '$switch': {
+                        'branches': [
+                            {'case': '$missing_field', 'then': 'first case'},
+                            {'case': True, 'then': '$missing_field'},
+                        ],
+                        'default': 'did not match',
+                    }
+                },
+            }},
+        ])
+        expected = {
+            '_id': 1,
+            'doc_example_1': 'less than',
+            'doc_example_2': 'did not match',
+            'doc_example_3': 'first case',
+        }
+        self.assertEqual([expected], list(actual))
+
+    def test__aggregate_switch_operation_failures(self):
+        self.db.collection.insert_one({'_id': 1, 'a': 0})
+
+        tests_cases = [
+            (
+                {'$switch': []},
+                '$switch requires an object as an argument, found: %s' % type([]),
+            ),
+            (
+                {'$switch': {}},
+                '$switch requires at least one branch.',
+            ),
+            (
+                {'$switch': {'branches': {}}},
+                "$switch expected an array for 'branches', found: %s" % type({}),
+            ),
+            (
+                {'$switch': {'branches': []}},
+                '$switch requires at least one branch.',
+            ),
+            (
+                {'$switch': {'branches': [{}, 7]}},
+                "$switch requires each branch have a 'case' expression"
+            ),
+            (
+                {'$switch': {'branches': [{'case': True}, 7]}},
+                "$switch requires each branch have a 'then' expression."
+            ),
+            (
+                {'$switch': {'branches': [{'case': True, 'then': 3}, 7]}},
+                '$switch expected each branch to be an object, found: %s' % type(0),
+            ),
+            (
+                {'$switch': {'branches': [7, {}]}},
+                '$switch expected each branch to be an object, found: %s' % type(0),
+            ),
+            (
+                {'$switch': {'branches': [{'case': False, 'then': 3}]}},
+                '$switch could not find a matching branch for an input, '
+                'and no default was specified.',
+            ),
+        ]
+
+        for switch_operator, expected_exception in tests_cases:
+            pipeline = [
+                {'$match': {'_id': 1}},
+                {'$project': {'result_field': switch_operator}},
+            ]
+            with self.assertRaises(mongomock.OperationFailure) as err:
+                self.db.collection.aggregate(pipeline)
+            self.assertEqual(expected_exception, str(err.exception))
```

Comment:
That (checking the exact exception string) seems like an overkill compared to other tests in here. Now that's it's here let's keep it, but if you do another change in mongomock, please only check the type of the error.